### PR TITLE
Make string.upper Unicode-aware

### PR DESCRIPTION
**Pull request by whiteh0le**
Problem
---
When using a language that has characters with diacritics or other special symbols, the function string.upper is not capable of converting then correctly, as demonstrated in the example below:
```lua
string.upper(Estás muerto)
-- returns "ESTáS MUERTO" and not "ESTÁS MUERTO" like it should
```

Proposed solution
---
Add `sh_datautf8.lua`, it contains the [case mappings](http://userguide.icu-project.org/transforms/casemappings) needed by the `utf8upper` and `utf8lower` functions, and replace all occurrences of `string.upper` by it's unicode-aware counterpart.

Demonstration
---
fix_unicode

### CODE REVIEW

**Comment by whiteh0le** [comment]
@impulsh Thanks for the merge! but there is a problem, the reason I named the new file `sh_datautf8.lua` and not `sh_utf8data` or something else is because it needs to load **before** sh_utf8.lua with the current name (sh_utf8casemap.lua) the framework won't load!

We have two options:
- Rename the file to make it load before `sh_utf8.lua`
- Include the file manually in `shared.lua` before we include the thirdparty libs
https://github.com/NebulousCloud/helix/blob/master/gamemode/shared.lua#L92

**Comment by impulsh** [comment]
Good catch. Relying on the filename for execution order isn't a good idea - including it manually is definitely the way to go

File: gamemode/core/libs/thirdparty/sh_utf8casemap.lua [hunk 1]
```diff
@@ -0,0 +1,1860 @@
...
+	["𐐟"] = "𐑇",
+	["𐐠"] = "𐑈",
+	["𐐡"] = "𐑉",
+	["𐐢"] = "𐑊",
+	["𐐣"] = "𐑋",
+	["𐐤"] = "𐑌",
+	["𐐥"] = "𐑍",
+	["𐐦"] = "𐑎",
+	["𐐧"] = "𐑏",
+}
```
Comment (ExtReMLapin) [inline gamemode/core/libs/thirdparty/sh_utf8casemap.lua:1859]: boi

Comment (whiteh0le) [reply gamemode/core/libs/thirdparty/sh_utf8casemap.lua:1859]: Is there anything wrong?😕

**Pull request merged**